crates/pathfinder/src/sequencer/request.rs

```diff
@@ -33,3 +33,273 @@ impl From<rpc::Call> for Call {
+            let mut ser =
+                serde_json::Serializer::with_formatter(Vec::new(), PythonDefaultFormatter);
+            parsed.serialize(&mut ser).unwrap();
+            let bytes = ser.into_inner();
+            let output = std::str::from_utf8(&bytes).expect("serde does this unchecked");
+
+            // these need to be byte for byte equal because we hash this
+            assert_eq!(input, output);
+        }
+    }
+
+    #[cfg(test)]
+    mod test_serde_features {
+        #[test]
+        fn serde_json_value_sorts_maps() {
+            // this property is leaned on and the default implementation of serde_json works like
+            // this. serde_json has a feature called "preserve_order" which could get enabled by
+            // accident, and it would destroy the ability to compute_contract_hash.
+
+            let input = r#"{"foo": 1, "bar": 2}"#;
+            let parsed = serde_json::from_str::<serde_json::Value>(input).unwrap();
+            let output = serde_json::to_string(&parsed).unwrap();
+
+            assert_eq!(output, r#"{"bar":2,"foo":1}"#);
+        }
+
+        #[test]
+        fn serde_json_has_arbitrary_precision() {
+            // the json has 251-bit ints, python handles them out of box, serde_json requires
+            // feature "arbitrary_precision".
+
+            // this is 2**256 - 1
+            let input = r#"{"foo":115792089237316195423570985008687907853269984665640564039457584007913129639935}"#;
+
+            let output =
+                serde_json::to_string(&serde_json::from_str::<serde_json::Value>(input).unwrap())
+                    .unwrap();
+
+            assert_eq!(input, output);
+        }
+
+        #[test]
+        fn serde_json_has_raw_value() {
+            // raw value is needed for others but here for completness; this shouldn't compile if
+            // you the feature wasn't enabled.
+
+            #[derive(serde::Deserialize, serde::Serialize)]
+            struct Program<'a> {
+                #[serde(borrow)]
+                debug_info: Option<&'a serde_json::value::RawValue>,
+            }
+
+            let mut input = serde_json::from_str::<Program>(
+                r#"{"debug_info": {"long": {"tree": { "which": ["we dont", "care", "about", 0] }}}}"#,
+            ).unwrap();
+
+            input.debug_info = None;
+
+            let output = serde_json::to_string(&input).unwrap();
+
+            assert_eq!(output, r#"{"debug_info":null}"#);
+        }
+    }
+}
+
+pub mod add_transaction {
+    use std::collections::HashMap;
+
+    use crate::core::{ConstructorParam, ContractAddressSalt, TransactionVersion};
+    use crate::rpc::serde::{
+        CallParamAsDecimalStr, CallSignatureElemAsDecimalStr, ConstructorParamAsDecimalStr,
+        FeeAsHexStr, TransactionVersionAsHexStr,
+    };
+
+    use serde_with::serde_as;
+
+    use super::contract::{EntryPointType, SelectorAndOffset};
+    use super::*;
```

Comment:
What does this import?

Reply:
You mean `super::*`? That actually imports a few things: `ContractAddress`, `EntryPoint`, `CallParam`, `Fee` and `CallSignatureElem`.

Reply:
Oki if those enumerable I would of course prefer to have them there explicitly intead of `use super::*;`.

```suggestion
    use super::{CallParam, CallSignatureElem, ContractAddress, Fee, EntryPoint};
```